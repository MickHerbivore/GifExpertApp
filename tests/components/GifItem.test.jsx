import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas de GifItem', () => {

    const title = 'Naruto Shippuden';
    const url = 'https://media4.giphy.com/media/2y98KScHKeaQM/giphy.gif?cid=4523aefel0986zifubtkxvuq8l41qr8fcrr1p1iylhlxvlfj&ep=v1_gifs_search&rid=giphy.gif&ct=g';

    test('should evaluate snapshot', () => {

        const { container } = render( <GifItem title={ title } url={ url }/>);
        expect( container ).toMatchSnapshot();

    });

    test('should mostrar la imagen con el URL y ALT indicado', () => {

        render( <GifItem title={ title } url={ url } /> );

        const { src, alt } = screen.getByRole( 'img' );
        expect( src ).toBe( url );
        expect( alt ).toBe( title );

    });

    test('should mostrar el titulo en el componente', () => {

        render( <GifItem title={ title } url={ url } /> );
        expect( screen.getByText( title ) ).toBeTruthy();

    })

})