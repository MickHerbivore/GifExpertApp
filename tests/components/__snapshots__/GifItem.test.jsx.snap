// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifItem should evaluate snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Naruto Shippuden"
      src="https://media4.giphy.com/media/2y98KScHKeaQM/giphy.gif?cid=4523aefel0986zifubtkxvuq8l41qr8fcrr1p1iylhlxvlfj&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Naruto Shippuden
    </p>
  </div>
</div>
`;
